mirobo/vacuum.py

```diff
@@ -139,13 +139,20 @@ def timer(self) -> List[Timer]:
 
         return timers
 
-    def set_timer(self, details):
-        # how to create timers/change values?
-        # ['ts', 'on'] to enable
-        raise NotImplementedError()
-        # return self.send(
-        #     "set_timer", [["ts", ["cron_line", ["start_clean", ""]]]])
-        # return self.send("upd_timer", ["ts", "on"])
+    def add_timer(self, cron, command, parameters):
+        import time
+        ts = int(round(time.time() * 1000))
+        return self.send("set_timer", [
+            [str(ts), [cron, [command, parameters]]]
+        ])
+
+    def delete_timer(self, timer_id: int):
+        return self.send("del_timer", [str(timer_id)])
+
+    def update_timer(self, timer_id: int, mode):
+        if mode != "on" and mode != "off":
```

Comment:
You would suggest a "enum class" here. ;-)

Reply:
I would indeed, was thinking about whether to have it for a boolean here. I changed it by adding Vacuum.TimerState {On, Off} ;-)